torch/_jit_internal.py

```diff
@@ -226,6 +226,32 @@ def can_compile_class(cls):
     return all(has_code)
 
 
+def get_type_hint_captures(fn):
+    """
+    Get a dictionary containing type resolution mapping necessary to resolve types
+    for the annotations on 'fn'. These are not considered to be closed-over by fn
+    and must be obtained separately.
+
+    Arguments:
+        fn: A callable.
+    Returns:
+        A Dict[str, Any] containing a mapping from the literal annotations used on
+        fn to the Python objects they refer to.
+    """
+    captures = {}
+    signature = inspect.signature(fn)
+
+    for name, parameter in signature.parameters.items():
```

Comment:
does this also cover return annotations? maybe worth adding a test case.

Reply:
Let me add that. Also, I discovered that `str(signature.parameter['name'])` removes any aliases, so the type annotation that is displayed is the string representation of the type object, not the string used to annotate the parameter in the source code. So we will have to actually inspect the source and parse out the function declaration. I'm going to add that too and re-request review.